src/enable_variable_ops/ngraph_capture_variables.cc

```diff
@@ -57,34 +57,45 @@ Status CaptureVariables(Graph* graph, std::set<string> skip_these_nodes) {
           {"AssignSub", std::make_pair("NGraphAssignSub", ReplaceAssign)},
           {"VariableV2", std::make_pair("NGraphVariable", ReplaceVariable)}};
 
-  std::vector<Node*> replaced_nodes;
+  std::set<Node*> ref_list;
+  std::vector<Node*> nodes_to_capture;
   for (auto node : graph->op_nodes()) {
-    if (NGraphPlacementRequested(node)) {
-      auto itr = CAPTURE_REPLACE_OP_MAP.find(node->type_string());
-      if (itr != CAPTURE_REPLACE_OP_MAP.end()) {
-        NGRAPH_VLOG(1) << "Capturing: " << node->name();
-        Node* replacement;
-
-        // Create the replacement node
-        TF_RETURN_IF_ERROR((itr->second.second)(graph, node, &replacement,
-                                                node->name(), itr->second.first,
-                                                false, false, 0, false));
-
-        std::vector<const Edge*> edges;
+    // Check if the node is a VariableV2
+    if (node->type_string() == "VariableV2") {
+      NGRAPH_VLOG(4) << "Found Variable: " << node->name();
+      // Add the Variable node to the ref list
+      ref_list.insert(node);
+
+      // go over all the nodes leading from VariableV2 and store them
+      // in a list if they are ref type
+      StoreRefTypeOutputs(node, &ref_list);
+
+      if (ref_list.size()) {
+        nodes_to_capture.insert(nodes_to_capture.end(), ref_list.begin(),
```

Comment:
we can add the check here. Enter in nodes_to_capture only if type exists in Capture_OP_Map